packages/calypso-products/src/plans-list.tsx

```diff
@@ -456,6 +456,12 @@ import type {
 } from './types';
 import type { TranslateResult } from 'i18n-calypso';
 
+const experiments: { [ key: string ]: string } = {};
+// DO NOT import. Use `setPlansListExperiment` instead.
+export const setExperiment = ( experimentName: string, variation: string ): void => {
+	experiments[ experimentName ] = variation;
+};
+
```

Comment:
I guess it's possible that we may end up wanting to do similar thing for other parts of `calypso-products`. Would it make sense to migrate this logic to a separate file (call it `experiments.ts` maybe) and define/export the various setters there? 

and there could be several ways to structure the data there, maybe a single `experiments` object with nested properties, or individual ones for each context e.g. `plansExperiments`, etc.

Reply:
Great idea!
I pushed the experiments-related code to its [own file](https://github.com/Automattic/wp-calypso/pull/84610/files#diff-eb9fdc0aa493a476f0e66f774ff9a7250ba37300f7ea2f594dfc5ac3a5967339).
We should be able to reuse this for other sections by specifying different namespaces, together with their setters and getters.

Reply:
Nice! and Nicely done! 😎